src/entries/popup/pages/ImportFromMetamask/ImportFromMetamask.tsx

```diff
@@ -0,0 +1,792 @@
+import { hex } from 'chroma-js';
+import { motion } from 'framer-motion';
+import { PropsWithChildren, useEffect, useRef, useState } from 'react';
+import { hexToNumber, isAddress, isHex } from 'viem';
+import { Address, useMutation } from 'wagmi';
+import { z } from 'zod';
+
+import { getAccounts } from '~/core/keychain';
+import { i18n } from '~/core/languages';
+import { SupportedCurrencyKey, supportedCurrencies } from '~/core/references';
+import {
+  appSessionsStore,
+  currentCurrencyStore,
+  useRainbowChainsStore,
+} from '~/core/state';
+import { useContactsStore } from '~/core/state/contacts';
+import { currentThemeStore } from '~/core/state/currentSettings/currentTheme';
+import { useRainbowChainAssetsStore } from '~/core/state/rainbowChainAssets';
+import { walletNamesStore } from '~/core/state/walletNames';
+import { ChainId } from '~/core/types/chains';
+import { getDappHost, getDappHostname } from '~/core/utils/connectedApps';
+import { mapToRange } from '~/core/utils/mapToRange';
+import { isLowerCaseMatch } from '~/core/utils/strings';
+import {
+  Box,
+  Button,
+  Inline,
+  Separator,
+  Stack,
+  Symbol,
+  Text,
+} from '~/design-system';
+import { foregroundColorVars } from '~/design-system/styles/core.css';
+import { globalColors } from '~/design-system/styles/designTokens';
+import { delay } from '~/test/utils';
+
+import { importWithSecret } from '../../handlers/wallet';
+
+const t = {
+  chainId: z.string().transform((s, ctx) => {
+    const chainId = isHex(s) ? hexToNumber(s) : +s;
+    if (isNaN(chainId)) {
+      ctx.addIssue({ code: z.ZodIssueCode.custom, message: 'Invalid ChainId' });
```

Comment:
lang file?

Reply:
these are dev error messages, shouldn't go to the end user